packages/eventual-send/src/E.js

```diff
@@ -304,24 +304,69 @@ export default makeE;
 
 /**
  * The awaited return type of a function.
+ * For the eventual result of an E call, @see {EResult} or @see {ECallableReturn}
  *
  * @template {(...args: any[]) => any} T
  * @typedef {T extends (...args: any[]) => infer R ? Awaited<R> : never} EReturn
  */
 
 /**
- * @template {import('./types.js').Callable} T
+ * An eventual value where remotable objects are recursively mapped to Remote types
+ *
+ * @template T
+ * @typedef {Awaited<T>} EResult
+ */
+
+/**
+ * Experimental type mapping remotable objects to Remote types
+ *
+ * @template T
  * @typedef {(
- *   ReturnType<T> extends PromiseLike<infer U>                       // if function returns a promise
- *     ? T                                                            // return the function
- *     : (...args: Parameters<T>) => Promise<EReturn<T>>  // make it return a promise
+ *   0 extends (1 & T)
```

Comment:
I have no idea what this line is about. Also below. Whatever this is doing, I have not seen this trick before.

Reply:
This is detecting an `any` type, and bailing if it is. Otherwise `any` causes both branches of a ternary to be taken, and the result is the union of both sides.